plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -1121,3 +1125,86 @@ def process_main(self, item=None):
         if pipe.returncode != 0:
             raise PulpCodedException(error_codes.RPM0001, command='repoview', stdout=stdout,
                                      stderr=stderr)
+
+
+class RemoveOldRepodataStep(platform_steps.PluginStep):
+    """
+    Remove repodata files that are not in repomd and are older then 14 day.
+    """
+    def __init__(self, content_dir, **kwargs):
+        """
+        Initialize the step for remove old repodata
+
+        :param content_dir: The base directory of the repository.  This directory should contain
+                            the repodata directory
+        :type content_dir: str
+        """
+        super(RemoveOldRepodataStep, self).__init__(
+            constants.PUBLISH_REMOVE_OLD_REPODATA_STEP,
+            **kwargs)
+        self.description = _('Removing old repodata')
+        self.content_dir = content_dir
+
+    def is_skipped(self):
+        """
+        Check the repo for the config option to remove old repodata.
+        Skip clean up if retain_old_repodata is true
+
+        :returns: return if step should clean old repodata or not
+        :rtype: bool
+        """
+        return not self.get_config().get('remove_old_repodata', True)
+
+    def remove_repodata_file(self, repodata_file):
+        os.remove(repodata_file)
+        self.progress_successes += 1
+        msg = _('Removed outdated %s' % os.path.basename(repodata_file))
+        logger.info(msg)
+
+    def filter_old_repodata(self, files):
+        exp = re.compile("([a-z0-9A-Z]+)-([a-z]+)(\.xml|\.xml\.gz|\.sqlite.bz2)")
+        to_remove = {}
+        groupped = {}
+        for f in files:
+            fname = os.path.basename(f)
+            if fname in ["repodata.xml"]:
+                continue
+            matched = exp.match(fname)
+            if matched:
+                delta = datetime.datetime.today() -\
+                    datetime.datetime.fromtimestamp(os.path.getmtime(f))
```

Comment:
Does this depend on https://pulp.plan.io/issues/2783 being implemented first?  Without that, the mtime will always be equal to the time of the CopyDirectoryStep and so I expect this will never remove anything.